.github/workflows/integration_tests_app_ci.yml

```diff
@@ -154,6 +154,12 @@ jobs:
             --test ../build/app/outputs/apk/androidTest/prod/debug/app-prod-debug-androidTest.apk \
             --device model=Pixel2,version=30,locale=en,orientation=portrait \
             --timeout 3m
+    
+      # It can easily happen that a dependency changed but the .lock file is not
+      # updated. Or other cases where files are changed during a build.
+      # Therefore, fails this check if there are Git changes.
+      - name: Fail if there are Git diffs
+        run: git diff --exit-code --name-status
```

Comment:
@Jonas-Sander FYI: I added the `--name-status` so that only the file names are printed:

```
$ git diff --exit-code --name-status
M	app/pubspec.lock
M	app/pubspec.yaml
```

Reply:
I can add this in a new PR for the other checks

Reply:
> I can add this in a new PR for the other checks

I updated it in this PR.